### ISSUE TITLE
prediction

### ISSUE THREAD

**Issue by RenEjima** [dreamquark-ai/tabnet#296]
# prediction by TabNetClassifier 

I'm trying to use TabNet for binary classification of the breast cancer dataset, and it's probably working well during training, but the inference results are strange, predicting all 0s and all 1s. If there is anything strange about what I am doing, please point it out to me.

# Code
#TabNetPretrainer
unsupervised_model = TabNetPretrainer(
    optimizer_fn=torch.optim.Adam,
    optimizer_params=dict(lr=2e-2),
    mask_type='entmax' # "sparsemax"
)

unsupervised_model.fit(
    X_train=X_train,
    eval_set=[X_valid],
    pretraining_ratio=0.8,
)

clf = TabNetClassifier(
    optimizer_fn=torch.optim.Adam,
    optimizer_params=dict(lr=2e-2),
    scheduler_params={"step_size":10, # how to use learning rate scheduler
                      "gamma":0.9},
    scheduler_fn=torch.optim.lr_scheduler.StepLR,
    mask_type='sparsemax' # This will be overwritten if using pretrain model
)

clf.fit(
    X_train=X_train, y_train=y_train,
    eval_set=[(X_train, y_train), (X_valid, y_valid)],
    eval_name=['train', 'valid'],
    eval_metric=['auc'],
    from_unsupervised=unsupervised_model
)
y_proba = clf.predict(X_test)
print('y_test')
print(y_test)
print('test predict')
print(y_proba)

# out put
Device used : cpu
epoch 0  | loss: 23866.77734| val_0_unsup_loss: 3812411.25|  0:00:00s
epoch 1  | loss: 9503.08008| val_0_unsup_loss: 895354.6875|  0:00:00s
epoch 2  | loss: 8184.47412| val_0_unsup_loss: 371195.6875|  0:00:00s
epoch 3  | loss: 6572.32373| val_0_unsup_loss: 256657.10938|  0:00:00s
......
epoch 70 | loss: 5.00937 | val_0_unsup_loss: 5.81055 |  0:00:02s

Early stopping occurred at epoch 70 with best_epoch = 60 and best_val_0_unsup_loss = 5.51955
Best weights from best epoch are automatically used!
Device used : cpu
/Users/xxx/miniforge3/envs/python38/lib/python3.8/site-packages/pytorch_tabnet/abstract_model.py:97: UserWarning: Pretraining: mask_type changed from sparsemax to entmax
  warnings.warn(wrn_msg)
Loading weights from unsupervised pretraining
epoch 0  | loss: 0.61485 | train_auc: 0.84433 | valid_auc: 0.84549 |  0:00:00s
epoch 1  | loss: 0.56318 | train_auc: 0.84991 | valid_auc: 0.83449 |  0:00:00s
epoch 2  | loss: 0.52741 | train_auc: 0.8915  | valid_auc: 0.77894 |  0:00:00s
......
epoch 10 | loss: 0.30436 | train_auc: 0.66164 | valid_auc: 0.59491 |  0:00:00s

Early stopping occurred at epoch 10 with best_epoch = 0 and best_valid_auc = 0.84549
Best weights from best epoch are automatically used!
y_test
[0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 0 0 0 1 1 0 1 1 0 1 0 1 0 1 0 1 0 1
 0 1 0 0 1 0 1 1 0 1 1 1 0 0 0 0 1 1 1 1 1 1 0 0 0 1 1 0 1 0 0 0 1 1 0 1 0
 0 1 1 1 1 1 0 0 0 1 0 1 1 1 0 0 1 0 1 0 1 1 0 1 1 1 1 1 1 1 0 1 0 1 0 0 1
 0 0 1 1 1 1 1 1 1 1 1 0 1 0 1 1 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 0]
test predict
[1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1]

**Comment by eduardocarvp** [comment]
Hello @RenEjima ,

I think the problem here is you are using `clf.predict` instead of `clf.predict_proba`. If you have a look at the two functions [here](https://github.com/dreamquark-ai/tabnet/blob/8c3b7951642f62e7449bb95875b5265d4b89148e/pytorch_tabnet/tab_model.py#L72) you will see that while `predict` applies `argmax` to the inputs, `predict_proba` simply outputs the result from the network forward plus a softmax.

Let us know if you can get it working!

**Comment by RenEjima** [comment]
Dear @eduardocarvp ,

Thank you for your help! It's a strange looking ROC-curve, but it's probably working just fine.
![Figure_1](https://user-images.githubusercontent.com/74614598/118085759-debda380-b3fd-11eb-9b9b-7d8e880a3f4e.png)

**Comment by eduardocarvp** [comment]
Happy it is working now!
I'm closing this issue, but feel free to open again if needed.
